Source/It.cs

```diff
@@ -141,5 +141,35 @@ public static string IsRegex(string regex, RegexOptions options)
 			// But evaluated every time :)
 			return Match<string>.Create(value => re.IsMatch(value), () => It.IsRegex(regex, options));
 		}
+
+		/// <summary>
+		/// Allows for setting up open generic methods on Mock Fixtures
+		/// </summary>
+		public interface AnyType
+		{
+			/// <summary>
+			/// Will contain the specific object being compared inside an open generic method.
+			/// Used for checking inside It.Is&lt;It.AnyType&gt;(x => ((int)x.Object) == 2)
+			/// </summary>
+			object Object { get; set; }
+		}
+
+		internal class AnyTypeImplementation : AnyType
+		{
+			public object Object { get; set; }
+		}
```

Comment:
It is not possible to get `It.AnySubTypeOf<TBase>` working as a type parameter directly due to the following

- `It.AnySubTypeOf<SomeBaseClass>' cannot be used as type parameter 'TReturnArg' in the generic type or method '...some method...'. There is no implicit reference conversion from 'It.AnySubTypeOf<SomeBaseClass>' to 'SomeBaseClass'.

I am still getting this compiler error even if I provide an implicit converter:

```
public class AnySubTypeOf<T> : It.AnyType where T : class
{
	public static implicit operator T(AnySubTypeOf<T> t)
	{
		return null;
	}
}
```
- Also cannot derive from a type parameters e.g.:

```
public class AnySubTypeOf<T> : T
{
}
```

The best I can come up with is the following:

```
mock.SetupGeneric(x => x
		.GetWhere<SomeBaseClass, SomeBaseClass, SomeBaseClass>(It.IsAnySubTypeOf<SomeBaseClass>())
	)
    .Return(...)
```

Where the mock is mocking the following method:

```
TReturnArg GetWhere<TInputArg1, TArgInput2, TReturnArg>(TInputArg1 arg1)
	where TInputArg1 : SomeBaseClass
	where TArgInput2 : SomeBaseClass
	where TReturnArg : SomeBaseClass;
```

Would that be sufficient?

It means adding a new method(s) `SetupGeneric()` and then using `It.IsAnySubTypeOf<SomeBaseClass>` which has the above implicit conversion to subtype, in order to satisfy input args.